pkg/apis/core/validation/validation.go

```diff
@@ -2169,7 +2172,9 @@ func ValidatePersistentVolumeClaimUpdate(newPvc, oldPvc *core.PersistentVolumeCl
 	}
 
 	if validateStorageClassUpgrade(oldPvcClone.Annotations, newPvcClone.Annotations,
-		oldPvcClone.Spec.StorageClassName, newPvcClone.Spec.StorageClassName) {
+		oldPvcClone.Spec.StorageClassName, newPvcClone.Spec.StorageClassName) ||
+		validateStorageClassUpgradeFromNil(oldPvcClone.Annotations,
```

Comment:
The error thrown below will complain about immutable `Annotations[v1.BetaStorageClassAnnotation]`, while Spec.StorageClassName should be immutable. It needs to distinct errors.

Reply:
Done. Moved it below to have it's own error. Not sure if immutable field error would be ok, it does not hint about it being mutable under certain conditions. Do we need a new error?